### ISSUE TITLE
[Bug]: Search failed with error `segment lacks` after Milvus recover from querynode pod failure chaos

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#30298]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20240125-f2c0ead5
- Deployment mode(standalone or cluster):cluster 
- MQ type(rocksmq, pulsar or kafka): kafka   
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2024-01-25T21:11:53.897Z] [2024-01-25 21:11:02 - INFO - ci_test]: index info: [{'collection': 'Checker__VfOuP78L', 'field': 'float_vector', 'index_name': 'index__GPxtACKY', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}] (test_all_collections_after_chaos.py:87)

[2024-01-25T21:11:53.897Z] [2024-01-25 21:11:02 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)

[2024-01-25T21:11:53.897Z] [2024-01-25 21:11:02 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-01-25T21:11:53.897Z] [2024-01-25 21:11:02 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.02548056382489183, 0.11634764009074929, 0.13105862266492177, 0.028210619387451962, 0.028462094791349462, 0.028307773064842178, 0.029844790160380334, 0.039450309709039386, 0.12099459033571477, 0.04583574509048263, 0.11695235312587882, 0.01636979466014124, 0.09306580341723944, 0.14878617972717123......, kwargs: {} (api_request.py:62)

[2024-01-25T21:11:53.897Z] [2024-01-25 21:11:02 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=503, message=failed to search: segment lacks[segment=447274033697723002]; segment lacks[segment=447274033698124016]: channel not available[channel=by-dev-rootcoord-dml_8_447274033698123216v1])>, <Time:{'RPC start': '2024-01-25 21:11:02.433339', 'RPC error': '2024-01-25 21:11:02.435382'}> (decorators.py:134)

[2024-01-25T21:11:53.897Z] [2024-01-25 21:11:02 - ERROR - ci_test]: Traceback (most recent call last):

[2024-01-25T21:11:53.897Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 32, in inner_wrapper

[2024-01-25T21:11:53.897Z]     res = func(*args, **_kwargs)

[2024-01-25T21:11:53.897Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 63, in api_request

[2024-01-25T21:11:53.897Z]     return func(*arg, **kwargs)

[2024-01-25T21:11:53.897Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 792, in search

[2024-01-25T21:11:53.897Z]     resp = conn.search(

[2024-01-25T21:11:53.897Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 135, in handler

[2024-01-25T21:11:53.897Z]     raise e from e

[2024-01-25T21:11:53.897Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 131, in handler

[2024-01-25T21:11:53.897Z]     return func(*args, **kwargs)

[2024-01-25T21:11:53.897Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 170, in handler

[2024-01-25T21:11:53.897Z]     return func(self, *args, **kwargs)

[2024-01-25T21:11:53.897Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 110, in handler

[2024-01-25T21:11:53.897Z]     raise e from e

[2024-01-25T21:11:53.897Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 74, in handler

[2024-01-25T21:11:53.897Z]     return func(*args, **kwargs)

[2024-01-25T21:11:53.897Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 786, in search

[2024-01-25T21:11:53.897Z]     return self._execute_search(request, timeout, round_decimal=round_decimal, **kwargs)

[2024-01-25T21:11:53.897Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 728, in _execute_search

[2024-01-25T21:11:53.897Z]     raise e from e

[2024-01-25T21:11:53.897Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 721, in _execute_search

[2024-01-25T21:11:53.897Z]     check_status(response.status)

[2024-01-25T21:11:53.897Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/utils.py", line 58, in check_status

[2024-01-25T21:11:53.897Z]     raise MilvusException(status.code, status.reason, status.error_code)

[2024-01-25T21:11:53.897Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=503, message=failed to search: segment lacks[segment=447274033697723002]; segment lacks[segment=447274033698124016]: channel not available[channel=by-dev-rootcoord-dml_8_447274033698123216v1])>

[2024-01-25T21:11:53.897Z]  (api_request.py:45)

[2024-01-25T21:11:53.898Z] [2024-01-25 21:11:02 - ERROR - ci_test]: (api_response) : <MilvusException: (code=503, message=failed to search: segment lacks[segment=447274033697723002]; segment lacks[segment=447274033698124016]: channel not available[channel=by-dev-rootcoord-dml_8_447274033698123216v1])> (api_request.py:46)
```

### Milvus Log

failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-kafka-cron/detail/chaos-test-kafka-cron/10892/pipeline

log:
[artifacts-querynode-pod-failure-10892-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/14059700/artifacts-querynode-pod-failure-10892-server-logs.tar.gz)

cluster: 4am
namespace: chaos-testing
release name: querynode-pod-failure-10892

```

[2024-01-25T21:09:35.165Z] + kubectl get pods -o wide

[2024-01-25T21:09:35.166Z] + grep querynode-pod-failure-10892

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-etcd-0                                1/1     Running            0                 33m     10.104.27.209   4am-node31   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-etcd-1                                1/1     Running            0                 33m     10.104.19.94    4am-node28   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-etcd-2                                1/1     Running            0                 33m     10.104.31.207   4am-node34   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-kafka-0                               2/2     Running            0                 33m     10.104.19.98    4am-node28   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-kafka-1                               2/2     Running            0                 33m     10.104.27.210   4am-node31   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-kafka-2                               2/2     Running            0                 33m     10.104.25.12    4am-node30   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-kafka-exporter-5965b7dfdd-nktpl       1/1     Running            4 (33m ago)       33m     10.104.4.67     4am-node11   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-datacoord-5b6cd75764-6vtb6     1/1     Running            0                 33m     10.104.12.142   4am-node17   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-datanode-7b9644f79-4kp9w       1/1     Running            0                 33m     10.104.12.141   4am-node17   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-datanode-7b9644f79-xvvjj       1/1     Running            0                 33m     10.104.31.200   4am-node34   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-indexcoord-5d97d5fdf7-w5v6w    1/1     Running            0                 33m     10.104.12.143   4am-node17   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-indexnode-6f895c46d-6fkcs      1/1     Running            0                 33m     10.104.12.145   4am-node17   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-indexnode-6f895c46d-84sf6      1/1     Running            0                 33m     10.104.31.203   4am-node34   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-indexnode-6f895c46d-q2trl      1/1     Running            0                 33m     10.104.13.181   4am-node16   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-proxy-6df4c54549-n8gzz         1/1     Running            0                 33m     10.104.12.140   4am-node17   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-querycoord-776ccb6586-wnf75    1/1     Running            0                 33m     10.104.12.144   4am-node17   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-querynode-86dcd5ff6d-5v5mw     1/1     Running            3 (11m ago)       33m     10.104.25.7     4am-node30   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-querynode-86dcd5ff6d-r8jhc     1/1     Running            3 (11m ago)       33m     10.104.12.139   4am-node17   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-querynode-86dcd5ff6d-zmb7m     1/1     Running            3 (11m ago)       33m     10.104.13.180   4am-node16   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-milvus-rootcoord-86b78d68-rn4gn       1/1     Running            0                 33m     10.104.13.179   4am-node16   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-minio-0                               1/1     Running            0                 33m     10.104.27.206   4am-node31   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-minio-1                               1/1     Running            0                 33m     10.104.19.93    4am-node28   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-minio-2                               1/1     Running            0                 33m     10.104.31.208   4am-node34   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-minio-3                               1/1     Running            0                 33m     10.104.25.14    4am-node30   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-zookeeper-0                           1/1     Running            0                 33m     10.104.19.96    4am-node28   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-zookeeper-1                           1/1     Running            0                 33m     10.104.27.215   4am-node31   <none>           <none>

[2024-01-25T21:09:35.727Z] querynode-pod-failure-10892-zookeeper-2                           1/1     Running            0                 33m     10.104.25.13    4am-node30   <none>           <none>
```

**Comment by zhuwenxing** [comment]
It seems like a stable issue, querynode pod failure with pulsar mq also failed.
```

[2024-01-25T19:19:39.579Z] [2024-01-25 19:18:38 - INFO - ci_test]: index info: [{'collection': 'Checker__3T4soMrQ', 'field': 'float_vector', 'index_name': 'index__HENlnVSd', 'index_param': {'index_type': 'HNSW', 'metric_type': 'L2', 'params': {'M': 48, 'efConstruction': 500}}}] (test_all_collections_after_chaos.py:87)

[2024-01-25T19:19:39.579Z] [2024-01-25 19:18:38 - DEBUG - ci_test]: (api_request)  : [Collection.load] args: [None, 1, 180], kwargs: {} (api_request.py:62)

[2024-01-25T19:19:39.579Z] [2024-01-25 19:18:38 - DEBUG - ci_test]: (api_response) : None  (api_request.py:37)

[2024-01-25T19:19:39.579Z] [2024-01-25 19:18:38 - DEBUG - ci_test]: (api_request)  : [Collection.search] args: [[[0.10156684258074204, 0.060827077938198006, 0.07136033983637644, 0.14979845713296466, 0.03628401311626212, 0.027441993439477997, 0.027095317006133435, 0.09683510161210324, 0.029133881882852965, 0.01686874777891062, 0.0043166367136205945, 0.060847463035922686, 0.051338725016147224, 0.13335806527818......, kwargs: {} (api_request.py:62)

[2024-01-25T19:19:39.579Z] [2024-01-25 19:18:38 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=503, message=failed to search: segment lacks[segment=447272296995495750]; segment lacks[segment=447272296995495750]: channel not available[channel=by-dev-rootcoord-dml_8_447272296995695449v1])>, <Time:{'RPC start': '2024-01-25 19:18:38.211721', 'RPC error': '2024-01-25 19:18:38.216640'}> (decorators.py:134)

[2024-01-25T19:19:39.579Z] [2024-01-25 19:18:38 - ERROR - ci_test]: Traceback (most recent call last):

[2024-01-25T19:19:39.579Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 32, in inner_wrapper

[2024-01-25T19:19:39.579Z]     res = func(*args, **_kwargs)

[2024-01-25T19:19:39.579Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 63, in api_request

[2024-01-25T19:19:39.579Z]     return func(*arg, **kwargs)

[2024-01-25T19:19:39.579Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 792, in search

[2024-01-25T19:19:39.579Z]     resp = conn.search(

[2024-01-25T19:19:39.579Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 135, in handler

[2024-01-25T19:19:39.580Z]     raise e from e

[2024-01-25T19:19:39.580Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 131, in handler

[2024-01-25T19:19:39.580Z]     return func(*args, **kwargs)

[2024-01-25T19:19:39.580Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 170, in handler

[2024-01-25T19:19:39.580Z]     return func(self, *args, **kwargs)

[2024-01-25T19:19:39.580Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 110, in handler

[2024-01-25T19:19:39.580Z]     raise e from e

[2024-01-25T19:19:39.580Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 74, in handler

[2024-01-25T19:19:39.580Z]     return func(*args, **kwargs)

[2024-01-25T19:19:39.580Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 786, in search

[2024-01-25T19:19:39.580Z]     return self._execute_search(request, timeout, round_decimal=round_decimal, **kwargs)

[2024-01-25T19:19:39.580Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 728, in _execute_search

[2024-01-25T19:19:39.580Z]     raise e from e

[2024-01-25T19:19:39.580Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 721, in _execute_search

[2024-01-25T19:19:39.580Z]     check_status(response.status)

[2024-01-25T19:19:39.580Z]   File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/utils.py", line 58, in check_status

[2024-01-25T19:19:39.580Z]     raise MilvusException(status.code, status.reason, status.error_code)

[2024-01-25T19:19:39.580Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=503, message=failed to search: segment lacks[segment=447272296995495750]; segment lacks[segment=447272296995495750]: channel not available[channel=by-dev-rootcoord-dml_8_447272296995695449v1])>

[2024-01-25T19:19:39.580Z]  (api_request.py:45)

[2024-01-25T19:19:39.580Z] [2024-01-25 19:18:38 - ERROR - ci_test]: (api_response) : <MilvusException: (code=503, message=failed to search: segment lacks[segment=447272296995495750]; segment lacks[segment=447272296995495750]: channel not available[channel=by-dev-rootcoord-dml_8_447272296995695449v1])> (api_request.py:46)
```
failed job: https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-cron/detail/chaos-test-cron/10970/pipeline
log:
[artifacts-querynode-pod-failure-10970-server-logs.tar.gz](https://github.com/milvus-io/milvus/files/14059741/artifacts-querynode-pod-failure-10970-server-logs.tar.gz)

**Comment by ThreadDao** [comment]
@congqixia reproduce image: `master-20240130-fda6f8d0`
https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-kafka-cron/detail/chaos-test-kafka-cron/11091/pipeline/289

**Comment by congqixia** [comment]
Shall be fixed after #30258 merged
could you please verify? 
/assign @zhuwenxing

**Comment by zhuwenxing** [comment]
The `segment lack` issue did not happen.
`unrecoverable error` when searching still exists, but this is another issue https://github.com/milvus-io/milvus/issues/30366

so this issue can be closed
